Borrow hazard in XRSession::event_callback

https://book.servo.org/design-documentation/script/garbage-collection-and-refcell.html#garbage-collection-and-refcell

https://github.com/servo/servo/blob/2d075cbb6655ff0777668763920cd7ad6ca12de8/components/script/dom/webxr/xrsession.rs#L289-L291

As demonstrated by https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=f8d896609f543ffff4e21d0eaeea6a81, this looks like it should be safe but actually is not because the mutable borrow remains active within the loop body. Using [safe_borrow_mut](https://doc.servo.org/script_bindings/cell/struct.DomRefCell.html#method.safe_borrow_mut) makes the compiler complain:
```
error[E0502]: cannot borrow `*cx` as mutable because it is also borrowed as immutable
   --> components/script/dom/webxr/xrsession.rs:290:21
    |
289 |                 for promise in self.end_promises.safe_borrow_mut(cx.no_gc()).drain(..) {
    |                                ---------------------------------------------
    |                                |                                 |
    |                                |                                 immutable borrow occurs here
    |                                a temporary with access to the immutable borrow is created here ...
290 |                     promise.resolve_native(cx, &());
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
291 |                 }
    |                 - ... and the immutable borrow might be used here, when that temporary is dropped and runs the destructor for type `RefMut<'_, Vec<std::rc::Rc<dom::promise::promise::Promise>>>`

For more information about this error, try `rustc --explain E0502`.
```

The easiest way to rewrite this is to use `self.end_promises.borrow().iter()` and then `self.end_promises.safe_borrow_mut(cx.no_gc()).clear()` afterwards.

--- a/components/script/dom/webxr/xrsession.rs
+++ b/components/script/dom/webxr/xrsession.rs
@@ -286,9 +286,10 @@ impl XRSession {
                 // Step 5: We currently do not have any such promises
                 // Step 6 is happening n the XR session
                 // https://immersive-web.github.io/webxr/#dom-xrsession-end step 3
-                for promise in self.end_promises.borrow_mut().drain(..) {
+                for promise in self.end_promises.borrow().iter() {
                     promise.resolve_native(cx, &());
                 }
+                self.end_promises.safe_borrow_mut(cx.no_gc()).clear();
                 // Step 7
                 let event = XRSessionEvent::new(
                     cx,

